runtimebp/cpu.go

```diff
@@ -42,13 +45,14 @@ func NumCPU() float64 {
 		return n
 	}
 
-	// Fallback to the standard runtime package value.
+	// If nothing is set, falling back to the core count is probably dangerous.
+	// Instead, fall back to the lowest value that still allows parallelism.
 	fmt.Fprintf(
 		os.Stderr,
-		"runtimebp.NumCPU: Failed to read cgroup v1, fallback to NumCPU on the physical machine: %v\n",
+		"runtimebp.NumCPU: Failed to read cgroup v1, fallback to 2 to allow parallelism: %v\n",
```

Comment:
and if we change `2` to a constant that should be a `%d` instead of hardcoding in the log message